oshi-core/src/test/java/oshi/util/MemoizerTest.java

```diff
@@ -150,6 +144,21 @@ public Long get() {
          * internal s.value was before this call increments it.
          */
         final long actualNumberOfIncrements = s.get() - 1;
+        testIncrementCounts(actualNumberOfIncrements, iterationDurationNanos, ttlNanos);
+
+    }
+
+    private void finishAllThreads(Collection<Future<Void>> results) throws Throwable {
+        for (final Future<Void> result : results) {
+            try {
+                result.get();
+            } catch (final ExecutionException e) {
```

Comment:
Now that this is not a part of the original method anymore, it makes even more sense to no unwrap the original exception in order to not lose the whole trace leading not only to the executor thread but also to this `finishAllThreads` method. I think my original attempt to make the stack trace shorter has the potential of reducing clarity rather than improving it (in other words, I did a stupid thing originally).

Reply:
So I should not catch exceptions here?

Reply:
Yep, just let it be thrown without unwrapping:
```java
private static void finishAllThreads(Collection<Future<Void>> results) throws ExecutionException, InterruptedException {
    for (final Future<Void> result : results) {
        result.get();
    }
}
```
Also, if a method is not designed to use instance variables (fields) and or be overridden, it always makes sense to make it static.

Reply:
Notice also how `expirationNanos ` is compared with `now ` in `Memoizer.memoize`:
```
long nanos = expirationNanos;
now - nanos >= 0
```
A delta is compared with 0 rather than just comparing `now >= nanos`. That's not just for fun :)